src/util/generate-token.js

```diff
@@ -0,0 +1,35 @@
+import { installTimeStorageKey } from 'src/imports/background'
+
+const API_PATH = '/user-token'
+
+export const USER_ID = 'user-id'
+
+export default async function generateTokenIfNot(installTime) {
+    const userId = (await browser.storage.local.get(USER_ID))[USER_ID]
+
+    // if install time is not present then current time stores as install time and send to server same
+    if (!installTime) {
+        installTime = Date.now()
+        browser.storage.local.set({ [installTimeStorageKey]: installTime })
+    }
+
+    const host =
+        'https://a8495szyaa.execute-api.eu-central-1.amazonaws.com/' +
+        (process.env.NODE_ENV === 'production' ? 'production' : 'staging')
+
+    if (!userId) {
+        const generateToken = await fetch(host + API_PATH, {
+            method: 'POST',
+            headers: {
+                'Content-type':
```

Comment:
~~We still need to think about some form of error handling here. To re-cap what we talked about a few weeks back I think the main solutions we had were (relating to a user not having an ID, either via network error or local storage loss):~~
~~1. create an alarm to regularly check the user ID storage key and, if empty, request one from the server~~
~~2. run a user ID storage checking hook before every request to the analytics API, and request a new ID if empty~~

~~1 would probably be the most straight-forward to implement now. 2 would require integration in #336~~

**EDIT: looking over #336 now and it seems to be already there :)**

Also 99% sure the spec is case-insensitive, after googling it, but I would still put this header as `'Content-Type'` to avoid issues with weird server implementations and just general consistency.